service/src/test/java/bio/terra/externalcreds/dataAccess/SshKeyPairDaoTest.java

```diff
@@ -0,0 +1,145 @@
+
+      assertPresent(loadedKeyOne);
+      assertPresent(loadedKeyTwo);
+      verifySshKeyPair(loadedKeyOne.get());
+      assertEquals(userId, loadedKeyTwo.get().getUserId());
+      assertEquals(externalUserEmail, loadedKeyTwo.get().getExternalUserEmail());
+      assertEquals(DEFAULT_KEY_TYPE, loadedKeyTwo.get().getType());
+      assertEquals(DEFAULT_PUBLIC_KEY, loadedKeyTwo.get().getPublicKey());
+      assertEquals(DEFAULT_PRIVATE_KEY, loadedKeyTwo.get().getPrivateKey());
+    }
+
+    @Test
+    void upsertSshKey_addId() {
+      var storedSshKey = sshKeyPairDAO.upsertSshKey(createDefaultSshKey());
+
+      verifySshKeyPair(storedSshKey);
+    }
+  }
+
+  @Nested
+  class GetSshKeyTest {
+    @Test
+    void getSshKey_noUserId() {
+      var empty = sshKeyPairDAO.getSecret("", DEFAULT_KEY_TYPE);
+
+      assertEmpty(empty);
+    }
+
+    @Test
+    void getSshKey() {
+      var sshKey = createDefaultSshKey();
+      sshKeyPairDAO.upsertSshKey(sshKey);
+
+      var loadedSshKeyOptional = sshKeyPairDAO.getSecret(DEFAULT_USER_ID, DEFAULT_KEY_TYPE);
+
+      assertPresent(loadedSshKeyOptional);
+      verifySshKeyPair(loadedSshKeyOptional.get());
+    }
+  }
+
+  @Nested
+  class DeleteKeyTest {
+    @Test
+    void deleteSshKey_successfullyDeleted() {
+      sshKeyPairDAO.upsertSshKey(createDefaultSshKey());
+
+      assertTrue(sshKeyPairDAO.deleteSecret(DEFAULT_USER_ID, DEFAULT_KEY_TYPE));
+
+      assertEmpty(sshKeyPairDAO.getSecret(DEFAULT_USER_ID, DEFAULT_KEY_TYPE));
+      assertFalse(sshKeyPairDAO.deleteSecret(DEFAULT_USER_ID, DEFAULT_KEY_TYPE));
+    }
+
+    @Test
+    void deleteSshKey_keyNotExist_returnsFalse() {
+      assertFalse(sshKeyPairDAO.deleteSecret(DEFAULT_USER_ID, DEFAULT_KEY_TYPE));
+    }
+
+    @Test
+    void deleteSshKey_specifyWrongType_returnsFalse() {
+      sshKeyPairDAO.upsertSshKey(createDefaultSshKey());
+
+      assertFalse(sshKeyPairDAO.deleteSecret(DEFAULT_USER_ID, SshKeyPairType.AZURE));
+
+      assertTrue(sshKeyPairDAO.deleteSecret(DEFAULT_USER_ID, DEFAULT_KEY_TYPE));
+    }
+  }
+
+  private static SshKeyPair createDefaultSshKey() {
+    return createSshKey(DEFAULT_USER_ID, DEFAULT_KEY_TYPE, DEFAULT_EXTERNAL_USER_EMAIL);
+  }
+
+  private static SshKeyPair createSshKey(
+      String userId, SshKeyPairType keyType, String externalUserEmail) {
+    return new SshKeyPair.Builder()
+        .type(keyType)
+        .privateKey(DEFAULT_PRIVATE_KEY)
+        .publicKey(DEFAULT_PUBLIC_KEY)
+        .userId(userId)
+        .externalUserEmail(externalUserEmail)
+        .build();
+  }
+
+  private void verifySshKeyPair(SshKeyPair storedSshKey) {
+    assertEquals(DEFAULT_USER_ID, storedSshKey.getUserId());
+    assertEquals(DEFAULT_KEY_TYPE, storedSshKey.getType());
+    assertEquals(DEFAULT_EXTERNAL_USER_EMAIL, storedSshKey.getExternalUserEmail());
+    assertEquals(DEFAULT_PRIVATE_KEY, storedSshKey.getPrivateKey());
+    assertEquals(DEFAULT_PUBLIC_KEY, storedSshKey.getPublicKey());
+  }
```

Comment:
this function should not be necessary, you should be able to do `assertEquals(expectedSshKeyPair, actualSshKeyPair)`